pkg/reconciler/namespace/namespace.go

```diff
@@ -269,9 +280,35 @@ func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string
 		c.enqueueGVR(gvr)
 	}
 
+	// Only record the scheduling decision once enqueueing was
+	// successful to ensure that requeueing for a given scheduling
+	// decision will be retried until successful.
+	c.setNamepaceContentsEnqueuedFor(ns)
+
 	return nil
 }
 
+// namepaceContentsEnqueuedFor retrieves the last scheduling decision
+// for which the contents of the given namespace were successfully
+// enqueued.
+func (c *Controller) namepaceContentsEnqueuedFor(ns *corev1.Namespace) (string, bool) {
+	key := clusters.ToClusterAwareKey(ns.ClusterName, ns.Name)
+	c.namespaceContentsEnqueuedForLock.RLock()
+	defer c.namespaceContentsEnqueuedForLock.RUnlock()
+	scheduling, ok := c.namespaceContentsEnqueuedForMap[key]
+	return scheduling, ok
+}
```

Comment:
these funcs with local locking make me nervous that we might get races due to multiple go routines influencing each other? E.g. the `DeleteFunc` definitely runs outside of the workers.

Reply:
One could move the `DeleteFunc` logic into the reconciler too. Per ns key there is only one go routine operating on it at a given time.

Reply:
DeleteFunc has locking - is this comment outdated?

Reply:
Please restate your request, I don't understand what action is implied.